lib/cog/repository/event_hooks.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Cog.Repository.EventHooks do
+  @moduledoc """
+  Behavioral API for interacting with event hooks. Prefer these
+  functions over direct manipulation with `Cog.Repo`.
+  """
+
+  alias Cog.Repo
+  alias Cog.Models.EventHook
+
+  @doc """
+  Retrieve the definition of the given event hook. The given id must
+  be a valid UUID.
+  """
+  @spec hook_definition(String.t) :: {:ok, %EventHook{}} | {:error, :not_found | :bad_id}
+  def hook_definition(id) do
+    if Cog.UUID.is_uuid?(id) do
+      case Repo.get(EventHook, id) do
+        %EventHook{} = hook ->
+          {:ok, hook}
+        nil ->
+          {:error, :not_found}
+      end
+    else
+      {:error, :bad_id}
+    end
+  end
+
+  def hook_definition!(id),
+    do: Repo.get!(EventHook, id)
```

Comment:
No, it's not... looks like that was something I tried but ended up backing out of. I'll remove it.